src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -3026,13 +3026,13 @@ private bool CheckInvocationArgMixingWithUpdatedRules(
                 }
             }
 
-            inferDeclarationExpressionValEscape();
+            inferDeclarationExpressionValEscape(argsOpt, localScopeDepth, escapeValues);
 
             mixableArguments.Free();
             escapeValues.Free();
             return valid;
 
-            void inferDeclarationExpressionValEscape()
+            void inferDeclarationExpressionValEscape(ImmutableArray<BoundExpression> argsOpt, SafeContext localScopeDepth, ArrayBuilder<EscapeValue> escapeValues)
```

Comment:
:memo: I was debugging in this area as part of this PR and noticed we're doing some captures